common/encryption/encrypt.h

```diff
@@ -52,12 +51,12 @@ std::vector<unsigned char*> encrypt_bytes(std::string model_data) {
         tag
     );
 
-    std::vector<unsigned char*> out_iv_tag{output, iv, tag};
-
-    return out_iv_tag;
+    *ciphertext = output;

```

Comment:
See below comment.

If we end up using a `unsigned char*` for the `ciphertext` argument, we'll have to index into the ciphertext array and `memcpy()` the output, IV, and tag.